2-Draft-Accepted/RFCxxxx-PSCredential-Improvements.md

```diff
@@ -0,0 +1,102 @@
+---
+RFC: RFCxxxx
+Author: Steve Lee
+Status: Draft
+SupercededBy: N/A
+Version: 0.1
+Area: Security
+Comments Due: June 30th, 2019
+Plan to implement: Yes, PS7
+---
+
+# Simplify and secure use of credentials in scripts
+
+Advanced scripts that touch many systems require multiple credentials and
+types of credentials particularly when orchestrating across different clouds.
+Best practice is to not hard code any credentials in scripts, but currently
+this requires custom code on different platforms to handle this securely.
+
+This RFC proposes new cmdlets to make this simpler and secure.
+
+## Motivation
+
+> As a PowerShell script author,
+> I can securely use multiple credentials,
+> so that I can automate complex orchestration across multiple remote resources.
+
+## User Experience
+
+Registering credentials for later use:
+
+```powershell
+Install-Module AzVault
+New-CredentialVault -Name Azure -Extension AzVault -Credential $azCred
+$cred = Get-Credential -AccessToken $token
+Register-Credential -Credential $cred -Name AzDevOps -Vault Azure
+$azDevOpsCred = Get-Credential -Name AzDevOps
```

Comment:
I think overloading Get-Credential like this is confusing

Reply:
Perhaps we can use `Get-Secret` instead